### ISSUE TITLE
Stack too deep error without viaIR and 0% coverage

### ISSUE THREAD

**Issue by vladikopl01** [sc-forks/solidity-coverage#760]
Hello, need some help. Had the same problems with `stack too deep` error and `viaIR` set is `true` as someone already reported. Done everything as you suggested but nothing helps.

My `solcover.js` file
```javascript
module.exports = {
  skipFiles: ['BCDeimosContractRef.sol', 'ERC20Token.sol'],
  configureYulOptimizer: true,
  solcOptimizerDetails: {
    peephole: false,
    inliner: false,
    jumpdestRemover: false,
    orderLiterals: true,
    deduplicate: false,
    cse: false,
    constantOptimizer: false,
    yul: false,
  },
};
```

My `hardhat.config.ts` file
```typescript
const config: HardhatUserConfig = {
  solidity: {
    version: '0.8.15',
    settings: {
      optimizer: {
        enabled: true,
        runs: 200,
      },
      // viaIR: true,
    },
  },
  networks: {
    ...networksConfig,
    hardhat: {
      gas: 99999999,
      gasPrice: 20000000000,
      blockGasLimit: 999999999,
      allowUnlimitedContractSize: true,
    },
  },
  defaultNetwork: 'hardhat',
  typechain: {
    outDir: 'typechain',
    target: 'ethers-v5',
  },
  etherscan: etherscanConfig,
  gasReporter: {
    coinmarketcap: envConfig.coinmarketcapApiKey,
    currency: 'USD',
    showTimeSpent: true,
    enabled: envConfig.reportGas,
  },
};
```

I had solidity version 0.8.15 as you can see, hardhat version is ^2.11.2 and solidity-coverage version is ^0.8.2. Also, I have installed a typychain package with version ^8.1.0.

With such configuration I ran into such errors:
```
Version
=======
> solidity-coverage: v0.8.2

Instrumenting for coverage...
=============================

> BCDeimosContract.sol

Coverage skipped for:
=====================

> BCDeimosContractRef.sol
> ERC20Token.sol

Compilation:
============

CompilerError: Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON) while enabling the optimizer. Otherwise, try removing local variables.
    --> contracts/BCDeimosContract.sol:1122:36:
     |
1122 | SaleInfo storage saleInfo = _sales[saleId_].saleInfo;
     |                                    ^^^^^^^


Error in plugin solidity-coverage: HardhatError: HH600: Compilation failed

For more info run Hardhat with --show-stack-traces
```

When `viaIR` is turned to true it has this output:
```
Version
=======
> solidity-coverage: v0.8.2

Instrumenting for coverage...
=============================

> BCDeimosContract.sol

Coverage skipped for:
=====================

> BCDeimosContractRef.sol
> ERC20Token.sol

Compilation:
============

YulException: Variable var_amount_1136 is 1 slot(s) too deep inside the stack.


Error in plugin solidity-coverage: HardhatError: HH600: Compilation failed

For more info run Hardhat with --show-stack-traces
```

Then I tried to find a workaround for this and set `yul` to true in `solcOptimizerDetails` property in `solcover.js` file and had this result:
```
-----------------------|----------|----------|----------|----------|----------------|
File                   |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------------------|----------|----------|----------|----------|----------------|
 contracts\            |        0 |        0 |        0 |        0 |                |
  BCDeimosContract.sol |        0 |        0 |        0 |        0 |... 919,920,921 |
-----------------------|----------|----------|----------|----------|----------------|
All files              |        0 |        0 |        0 |        0 |                |
-----------------------|----------|----------|----------|----------|----------------|

> Istanbul reports written to ./coverage/ and ./coverage.json
```

> All examples were performed right after `npx hardhat clean` and `npx hardhat coverage` commands

**Comment by rkdud007** [comment]
I had same issue. I solved this by using these options
https://github.com/ethereum/solidity/issues/11638#issuecomment-1301656139

**Comment by cathalmf** [comment]
Im having this problem too. No other optimizer options work. 

Without enabling viaIR i get "stack too deep", so i dont have a choice.

**Comment by jmendiola222** [comment]
Same problem here, see my comment on https://github.com/sc-forks/solidity-coverage/issues/715#issuecomment-1369770303

**Comment by SevenSwen** [comment]
I had same issue. 
https://github.com/1inch/farming/tree/feature/enable_viaIR 
If using [this scheme](https://github.com/sc-forks/solidity-coverage/issues/715#issuecomment-1431239520) then all tests into coverage will crash with 
`NomicLabsHardhatPluginError: The contract <contract_name> is missing links for the following libraries: @openzeppelin/contracts/utils/math/Math.sol:Math`
I think this happens because we have remove `i` flag (full inliner). But if we will return inliner (change "gvf" step to "gvif") than the error will disappear, but the coverage will still remain 0%.

It seems to me that turning off the inliner should not affect the coverage so much that the internal libraries are no longer inlined. In normal tests, I could not reproduce such logic, so it is probably the coverage that affects this case.

**Comment by emretepedev** [comment]
same issue here.

**Comment by diego-G** [comment]
Same problem here...

**Comment by hung-native** [comment]
same here

**Comment by diego-G** [comment]
It doesn't @vladikopl01. ViaIR is necessary to compile due to Stack too deep errors. Then we loose the coverage.

**Comment by jmendiola222** [comment]
There is a proposed workaround [here](https://github.com/sc-forks/solidity-coverage/issues/715#issuecomment-1431239520), but it seems it only worked for one user, certainly not for me. :/

**Comment by mariogutval** [comment]
> It doesn't @vladikopl01. ViaIR is necessary to compile due to Stack too deep errors. Then we lose the coverage.

Same issue here

**Comment by zarifpour** [comment]
We were able to resolve our need for `viaIR` by using remix to connect with our localhost to identify the problematic contract(s) causing the `Stack too deep.` error. 

Once our workspace was imported into remix, we manually compiled each contract and monitored the output of the compiler. We identified a single mock file causing the issue. When the mock was removed and we disabled `viaIR` our coverage was back as expected.

It would be awesome if errors and warnings are isolated to the individual file(s) causing them. That would have saved us a lot of time.

I hope this helps!
